### DISCUSSION TITLE
e-label with a static content

### DISCUSSION THREAD

**Discussion by susedv** [filipecarneiro/ELabel#17] [Ideas]
The way the e-label information is presented is not appropriate for our customer in Germany. Maybe there are different requirements in different countries. The idea is to add a possibility to use a static e-label content instead of the generated tables of ingredients. This could be just a HTML field named "Content". If it is empty, then the content will be generated, otherwise this input will be shown. Similarly there may be a HTML field named "Footer" (as requested in the topic "Impressum Section").

Here is what our customer expects to be:

![grafik](https://github.com/filipecarneiro/ELabel/assets/43996715/99369018-bf7e-4c1e-8009-66bd255a5c35)

Description:

1)

> Contains small amounts of: fat,
saturated fat, acids, protein and salt

Such ingredients of small amounts should not appear in the table.

2) 

Current representation:

> Grapes, Tirage liqueur, Expedition liqueur, Preservative (**Sulphites**), Stabiliser (**Carboxymethylcellulose**)

Required representation:

> Required: Grapes, Tirage liqueur, Expedition liqueur, Preservative: **Sulphites**, Stabiliser: Carboxymethylcellulose

- Only "Sulphites" may be bold - all other igredients should be regular
- Use colons instead of brackets

3) Nutritions per 100 ml: it looks like it is currently calculated per 1000 ml (or am I wrong?)

A custom HTML content field may be used to solve problems like this. What do you think?

**Comment by filipecarneiro** [comment]
Hi @susedv 
I think that all your requirements could be accomplished with some additional configuration (simple and useful, in fact).

A custom HTML content would miss all the benefits of having a database, automatic calculations and translations. In this case, it would be better to have a static site…

Regarding "Carboxymethylcellulose" as not bold, can you please check if that ingredient is configured as Allergen? If yes, just remove that flag (go to /Admin/Ingredient?filterText=Carboxymethylcellulose)

Regarding nº 3, can you share detailed data of your case?

Let me think about configuration for nº 1 and nº 2.

**Comment by filipecarneiro** [comment]
For nº 1: Nutrition declaration table without negligible nutrients #18

**Comment by filipecarneiro** [comment]
For nº 2: Ingredient category presentation format options #19

**Comment by susedv** [comment]
Our customer requested so many changes that I started to fork this project. Of cource I would share my modifications with you, but I think you don't want to have them in the main branch.

Currently I did it as follows:

1) If Fat, Salt or Protein is below 1.0, then show "Contains small amounts of: fat,
saturated fat, acids, protein and salt" in the last table row instead of column based appearance.

2.1) Yes, solved: Allergen was set for Carboxymethylcellulose, maybe the user mistook or it is different for various countries.

2.2) Using colons instead of parentheses (Catergory: Ingredient). Somebody said to our customer that it is required, I don't know.

3) Sorry our customer mistook, I think it looks good (similar to the image above: 70kcal = 293 kJ).

But our customer requested even more changes, I will show you when I have finished them.

**Reply by filipecarneiro** [reply]
Please note that the tolerance for Fat, Carbohydrate, Sugars and Protein is **0.5g**/100g. For Saturates is **0.1g**/100g and Salt is **0.0125g**/100g.
See the [EU guidance table](https://food.ec.europa.eu/system/files/2016-10/labelling_nutrition-vitamins_minerals-guidance_tolerances_summary_table_012013_en.pdf).

**Reply by susedv** [reply]
Thank you for this comment, I will improve it later. I did it this way as a "dirty hack", because it is needed quickly for a single product, but later it should be solved properly, I hope the main branch will include it. 

Here are the modified views:

[Views.zip](https://github.com/filipecarneiro/ELabel/files/14462655/Views.zip)

**Comment by susedv** [comment]
Here is what I did because our customer requested it:

![grafik](https://github.com/filipecarneiro/ELabel/assets/43996715/bca73d0d-70b6-4d7e-8a8d-25acbaa439aa)

**Comment by filipecarneiro** [comment]
Apart from the last line, that can be accomplished with #18 , I think that all the alignments and titles with colons could be easily added with a custom CSS.

Actually, even the list of ingredients with colons instead of brackets (#19), could be made with custom CSS if we add more classes to HTML.

**Comment by filipecarneiro** [comment]
This leads me to a new feature request :)

Allow label styling with custom CSS #20
